sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/DateTimeBenchmark.scala

```diff
@@ -114,12 +114,12 @@ object DateTimeBenchmark extends SqlBasedBenchmark {
     }
     runBenchmark("Parsing") {
       val n = 1000000
-      val timestampStrExpr = "concat('2019-01-27 11:02:01.', cast(mod(id, 1000) as string))"
+      val timestampStrExpr = "concat('2019-01-27 11:02:01.', rpad(mod(id, 1000), 3, '0'))"
       val pattern = "'yyyy-MM-dd HH:mm:ss.SSS'"
```

Comment:
one more thought: maybe we should change the pattern string to `yyyy-M-d H:m:s.S`, to match the behavior of 2.4

Reply:
It will fail with `2019-01-27 11:02:01.12(123)`